parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -125,6 +126,18 @@ public class Parquet {
 
   private Parquet() {}
 
+  public static void register() {
+    InternalData.register(FileFormat.PARQUET, Parquet::writeInternal, Parquet::readInternal);
```

Comment:
We should move this up into the InternalData initialization.  I think we want to explicitly register, not rely on self registration.

Reply:
I wanted to reduce the amount of reflection code so I thought it would make sense to have the majority of this code in the format. We could load the `writeInternal` and `readInternal` methods dynamically, I guess.